memory/iter.go

```diff
@@ -28,19 +28,45 @@ type iterator struct {
 	// reset is a mutatuble property that indicates whether the iterator should be
 	// returned to the beginning on the next [Next] call.
 	reset bool
+
+	// These properties are used by a work around for a bug in the btree implementation:
+	// https://github.com/tidwall/btree/issues/46 - these properties and the work around
+	// should be removed when the btree bug is fixed.
+	//
+	// `TestBTreePrevBug` also documents this issue.
+	lastItem  dsItem
+	firstItem dsItem
```

Comment:
todo: Please add a documentation that this is only needed for `reverse` operations.

Reply:
Will do.

- [ ] Note is for reverse only